pajbot/apiwrappers/twitch/badges.py

```diff
@@ -0,0 +1,124 @@
+import logging
+
+from pajbot.apiwrappers.response_cache import ClassInstanceSerializer
+from pajbot.apiwrappers.base import BaseAPI
+
+log = logging.getLogger(__name__)
+
+
+class BadgeNotFoundError(ValueError):
+    pass
+
+
+class BadgeVersion:
+    def __init__(self, image_url_1x, image_url_2x, image_url_4x, description, title):
+        self.image_url_1x = image_url_1x
+        self.image_url_2x = image_url_2x
+        self.image_url_4x = image_url_4x
+        self.description = description
+        self.title = title
+
+    def jsonify(self):
+        return {
+            "image_url_1x": self.image_url_1x,
+            "image_url_2x": self.image_url_2x,
+            "image_url_4x": self.image_url_4x,
+            "description": self.description,
+            "title": self.title,
+        }
+
+    @staticmethod
+    def from_json(json_data):
+        image_url_1x = json_data["image_url_1x"]
+        image_url_2x = json_data["image_url_2x"]
+        image_url_4x = json_data["image_url_4x"]
+        description = json_data["description"]
+        title = json_data["title"]
+        return BadgeVersion(
+            image_url_1x=image_url_1x,
+            image_url_2x=image_url_2x,
+            image_url_4x=image_url_4x,
+            description=description,
+            title=title,
+        )
+
+
+class BadgeSet:
+    def __init__(self, set_id, versions):
+        self.set_id = set_id
+        self.versions = versions
+
+    def jsonify(self):
+        return {
+            "set_id": self.set_id,
+            "versions": {
+                badge_version_id: badge_version.jsonify() for badge_version_id, badge_version in self.versions.items()
+            },
+        }
+
+    @staticmethod
+    def from_json(json_data):
+        set_id = json_data["set_id"]
+        versions = {
+            badge_version_id: BadgeVersion.from_json(badge_version_json)
+            for badge_version_id, badge_version_json in json_data["versions"].items()
+        }
+
+        return BadgeSet(set_id=set_id, versions=versions)
+
+
+class BadgeSets(dict):
+    def jsonify(self):
+        return {badge_set_id: badge_set.jsonify() for badge_set_id, badge_set in self.items()}
+
+    @staticmethod
+    def from_json(json_data):
+        sets = {badge_set_id: BadgeSet.from_json(badge_set_json) for badge_set_id, badge_set_json in json_data.items()}
+
+        return BadgeSets(sets)
+
+
+# https://discuss.dev.twitch.tv/t/beta-badge-api/6388
+class TwitchBadgesAPI(BaseAPI):
+    def __init__(self, redis):
+        super().__init__(base_url="https://badges.twitch.tv/v1/badges", redis=redis)
```

Comment:
I would make the base URL `https://badges.twitch.tv/v1/`, also note the trailing slash

Reply:
Ready for re-review